src/Behat/Behat/Context/Reader/AnnotatedContextReader.php

```diff
@@ -140,9 +143,18 @@ private function readDocBlockCallees($class, ReflectionMethod $method, $docBlock
                 continue;
             }
 
-            if ($callee = $this->readDocLineCallee($class, $method, $docLine, $description)) {
-                $callees[] = $callee;
+            if (!($callee = $this->readDocLineCallee($class, $method, $docLine, $description))) {
+                continue;
             }
+
+            if ($callee instanceof Definition) {
+                if (isset($patterns[$callee->getPattern()])) {
+                    continue;
+                }
+                $patterns[$callee->getPattern()] = true;
```

Comment:
I'm using `[$pattern => true]` & `isset()` since it should be faster than `in_array`, at the cost of readability.